### ISSUE TITLE
circular dependency has been detected with swagger plugin 

### ISSUE THREAD

**Issue by Nicolasvegam** [nestjs/swagger#2936]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

It throwed:

```
/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:184
            throw new Error(`A circular dependency has been detected (property key: "${key}"). Please, make sure that each side of a bidirectional relationships are using lazy resolvers ("type: () => ClassType").`);
                  ^
Error: A circular dependency has been detected (property key: "serviceCheckoutQuotes"). Please, make sure that each side of a bidirectional relationships are using lazy resolvers ("type: () => ClassType").
    at SchemaObjectFactory.createNotBuiltInTypeReference (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:184:19)
    at SchemaObjectFactory.createSchemaMetadata (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:294:25)
    at SchemaObjectFactory.mergePropertyWithMetadata (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:131:21)
    at /Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:82:35
    at Array.map (<anonymous>)
    at SchemaObjectFactory.extractPropertiesFromType (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:81:52)
    at SchemaObjectFactory.exploreModelSchema (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:103:41)
    at /Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:36:36
    at Array.map (<anonymous>)
    at SchemaObjectFactory.createFromModel (/Users/nicolasvega/Desktop/carvuk/carvuk-api-nest-js/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:20:45)
```

The nest-cli

```
{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "plugins": ["@nestjs/swagger"]
  }
}

```

### Minimum reproduction code

https://privaterepo.com

### Expected behavior

Render the Swagger Docs

### Package version

^7.3.1

### NestJS version

^10.0.0

### Node.js version

v21.7.2

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

yarn build is just fine.

**Comment by kamilmysliwiec** [comment]
Please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project).

**Comment by ausir0726** [comment]
I also encountered the same issue when I was implementing the official example at https://docs.nestjs.com/openapi/operations#advanced-generic-apiresponse. When using the cli-plugin and loading an example with @ApiExtraModel containing TData, it will cause a circular dependency.

The solution is to keep @ApiProperty() in the parameter of TData and not remove it.

I'm not sure if the original poster is facing the same issue, but I wanted to share my problem through this ISSUE.

here is minium reproduction
https://codesandbox.io/p/devbox/nifty-zhukovsky-k7ltzc

**Comment by LhonRafaat** [comment]
+1 one to this. I've used the swagger plugin + swc, I added my *dto.ts and *.type.ts files to dtoFileNameSuffix option. In couple of these files I have ` coordinates: [number, number];` which means two exactly have two elements inside the array, however Nestjs throws a circular dependency error with this format ` A circular dependency has been detected (property key: "coordinates").`. using `number[]` insterad fixes it, but I do not want to have array of numbers, but exactly TWO elements which represent coordinates.
